ipam/ippool_manager.go

```diff
@@ -154,6 +163,33 @@ func (m *IPPoolManager) getIndexes(ctx context.Context) (map[ipamv1.IPAddressStr
 		addresses[addressObject.Spec.Address] = claimName
 	}
 
+	// get list of IPAddress objects for cluster.x-k8s.io addresses
+	K8sAddressObjects := capipamv1.IPAddressList{}
```

Comment:
Keep it lower case to avoid confusion with public/exported variables. I would also prefer to name it something with CAPI rather than k8s.
```suggestion
	capiAddressObjects := capipamv1.IPAddressList{}
```